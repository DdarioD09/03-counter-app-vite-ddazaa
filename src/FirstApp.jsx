import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {

  return (
    <>
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      {/* <h2>{getGreting('Dario')}</h2> */}
      <h1>{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
  name: "Dario Daza",
  subTitle: "There is no subtitle",
  title: "There is no title",
}