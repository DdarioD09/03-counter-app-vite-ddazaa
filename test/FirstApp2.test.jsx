const { render, screen } = require("@testing-library/react")
import { FirstApp } from '../src/FirstApp'

describe('Tests in <FirstApp />', () => {

    const title = 'Hello, how are you';
    const subTitle = 'This is a test subtitle';

    test('should match with the snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('should show the message "Hello, how are you"', () => {
        render(<FirstApp title={title} />);
        expect(screen.getAllByText(title)).toBeTruthy();
        // screen.debug();
    });

    test('should show the title in a h1 tag', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('should show the subtitle sended by props', () => {
        render(
            <FirstApp
                title={title}
                subTitle={subTitle}
            />
        );
        expect(screen.getAllByText(subTitle).length).toBe(2);

    })
});