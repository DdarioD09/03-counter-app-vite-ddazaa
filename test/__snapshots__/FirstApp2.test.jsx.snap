// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hello, how are you
     
  </h1>
  <p>
    There is no subtitle
  </p>
  <p>
    There is no subtitle
  </p>
  <p>
    Dario Daza
  </p>
</div>
`;
